Crash in org.snakeyaml.engine.v2.api.LoadSettingsBuilder

There is a reproducible crash for org.snakeyaml.engine.v2.api.LoadSettingsBuilder (dependency of [kaml](https://github.com/charleskorn/kaml)).

```
java.lang.ExceptionInInitializerError
at org.snakeyaml.engine.v2.api.LoadSettingsBuilder.<init>(LoadSettingsBuilder.java:66)
at org.snakeyaml.engine.v2.api.LoadSettings.builder(LoadSettings.java:79)
at com.charleskorn.kaml.YamlParser.<init>(YamlParser.kt:33)
at com.charleskorn.kaml.Yaml.decodeFromReader(Yaml.kt:51)
at com.charleskorn.kaml.Yaml.decodeFromStream(Yaml.kt:47)
at de.westnordost.streetcomplete.ktx.ResourcesKt.getYamlStringMap(Resources.kt:16)
at de.westnordost.streetcomplete.about.ChangelogFragmentKt$readChangelog$2.invokeSuspend(ChangelogFragment.kt:93)
at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:39)
at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
Caused by: java.util.regex.PatternSyntaxException: Syntax error in regexp pattern near index 13:
^\$\{\s*((?<name>\w+)((?<separator>:?(-|\?))(?<value>\w+)?)?)\s*\}$
^
at java.util.regex.Pattern.compileImpl(Native Method)
at java.util.regex.Pattern.compile(Pattern.java:411)
at java.util.regex.Pattern.<init>(Pattern.java:394)
at java.util.regex.Pattern.compile(Pattern.java:381)
at org.snakeyaml.engine.v2.resolver.JsonScalarResolver.<clinit>(JsonScalarResolver.java:41)
... 15 more
```

The crash reports I got so far were both from Android 5.1 devies and both (so far) from people who installed the app from F-Droid.

cc @FloEdelmann

--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -105,9 +105,15 @@ repositories {
 }
 
 configurations {
-    // it's already included in Android
     all {
+        // it's already included in Android
         exclude(group = "net.sf.kxml", module = "kxml2")
+
+        resolutionStrategy.dependencySubstitution {
+            substitute(module("org.snakeyaml:snakeyaml-engine:2.3"))
+                .using(module("org.bitbucket.snakeyaml:snakeyaml-engine:8209bb9484"))
+                .because("https://github.com/streetcomplete/StreetComplete/issues/3889")
+        }
     }
 }
 

--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -109,6 +109,7 @@ configurations {
         // it's already included in Android
         exclude(group = "net.sf.kxml", module = "kxml2")
 
+        // TODO remove substitution when `kaml` dependency uses newer version of `org.snakeyaml:snakeyaml-engine`
         resolutionStrategy.dependencySubstitution {
             substitute(module("org.snakeyaml:snakeyaml-engine:2.3"))
                 .using(module("org.bitbucket.snakeyaml:snakeyaml-engine:8209bb9484"))

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -14,6 +14,7 @@ allprojects {
     repositories {
         google()
         mavenCentral()
+        maven { url = java.net.URI("https://jitpack.io") }
     }
 }
 

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -14,6 +14,7 @@ allprojects {
     repositories {
         google()
         mavenCentral()
+        // TODO remove when dependency to org.bitbucket.snakeyaml:snakeyaml-engine:8209bb9484 is no longer needed
         maven { url = java.net.URI("https://jitpack.io") }
     }
 }

